stdlib/public/core/SIMDVector.swift

```diff
@@ -1407,3 +1499,159 @@ where T: SIMD, T.Scalar: FloatingPoint {
+    -> (Self, (TangentVector) -> (TangentVector, TangentVector)) {
+    return (lhs * rhs, { v in
+      return (v * rhs, v * lhs)
+    })
+  }
+  
+  @inlinable
+  static func _vjpDivide(lhs: Self, rhs: Self)
+    -> (Self, (TangentVector) -> (TangentVector, TangentVector)) {
+    return (lhs / rhs, { v in
+      (v / rhs, -lhs / (rhs * rhs) * v)
+    })
+  }
+}
+
+extension SIMD
+  where Self : Differentiable,
+        TangentVector : SIMD,
+        Scalar : BinaryFloatingPoint & Differentiable,
+        Scalar.TangentVector : BinaryFloatingPoint,
+        TangentVector.Scalar == Scalar.TangentVector {
+  @inlinable
+  static func _vjpAdd(lhs: Scalar, rhs: Self)
+    -> (Self, (TangentVector) -> (Scalar.TangentVector, TangentVector)) {
+    return (lhs + rhs, { v in
+      return (v.sum(), v)
+    })
+  }
+
+  @inlinable
+  static func _vjpSubtract(lhs: Scalar, rhs: Self)
+    -> (Self, (TangentVector) -> (Scalar.TangentVector, TangentVector)) {
+    return (lhs - rhs, { v in
+      return (v.sum(), -v)
+    })
+  }
+
+  @inlinable
+  static func _vjpAdd(lhs: Self, rhs: Scalar)
+    -> (Self, (TangentVector) -> (TangentVector, Scalar.TangentVector)) {
+    return (lhs + rhs, { v in
+      return (v, v.sum())
+    })
+  }
+
+  @inlinable
+  static func _vjpSubtract(lhs: Self, rhs: Scalar)
+    -> (Self, (TangentVector) -> (TangentVector, Scalar.TangentVector)) {
+    return (lhs - rhs, { v in
+      return (v, -v.sum())
+    })
+  }
+}
+
+extension SIMD
+  where Self : Differentiable,
+        TangentVector : SIMD,
+        Scalar : BinaryFloatingPoint & Differentiable,
+        Self.TangentVector == Self,
+        Scalar.TangentVector == Scalar {
+  @inlinable
+  static func _vjpMultiply(lhs: Self, rhs: Scalar)
+    -> (Self, (TangentVector) -> (TangentVector, Scalar.TangentVector)) {
+    return (lhs * rhs, { v in
+      return (v * rhs, (v * lhs).sum())
+    })
+  }
+  
+  @inlinable
+  static func _vjpDivide(lhs: Self, rhs: Scalar)
+    -> (Self, (TangentVector) -> (TangentVector, Scalar.TangentVector)) {
+    return (lhs / rhs, { v in
+      (v / rhs, (-lhs / (rhs * rhs) * v).sum())
+    })
+  }
+  
+  @inlinable
+  static func _vjpMultiply(lhs: Scalar, rhs: Self)
+    -> (Self, (TangentVector) -> (Scalar.TangentVector, TangentVector)) {
+    return (lhs * rhs, { v in
+      return ((v * rhs).sum(), v * lhs)
+    })
+  }
+  
+  @inlinable
+  static func _vjpDivide(lhs: Scalar, rhs: Self)
+    -> (Self, (TangentVector) -> (Scalar.TangentVector, TangentVector)) {
+    return (lhs / rhs, { v in
+      ((v / rhs).sum(), -lhs / (rhs * rhs) * v)
+    })
+  }
+}
+
+extension SIMD
+  where Self : Differentiable,
+        TangentVector : SIMD,
+        Scalar : BinaryFloatingPoint & Differentiable,
+        Scalar.TangentVector : BinaryFloatingPoint,
+        TangentVector == Self {
+  @inlinable
+  func _vjpSum() -> (Scalar, (Scalar.TangentVector) -> TangentVector) {
+    return (sum(), { v in Self(repeating: Scalar(v)) })
```

Comment:
Similar to my comment below about the `Scalar` initializer call, I would need the more strict requirement of `Scalar.TangentVector == Scalar` instead of `Scalar.TangentVector : BinaryFloatingPoint`, or else I get the error
```
error: cannot convert return expression of type '(Self.Scalar) -> Self' to return type '(_) -> _'
    return (sum(), { v in Self(repeating: v) })
                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~
```